hub/core/storage/s3.py

```diff
@@ -209,6 +186,39 @@ def clear(self):
         else:
             super().clear()
 
+    def __getstate__(self):
+        return (
+            self.root,
+            self.aws_access_key_id,
+            self.aws_secret_access_key,
+            self.aws_session_token,
+            self.aws_region,
+            self.endpoint_url,
+            self.max_pool_connections,
+            self.expiration,
+            self.tag,
+            self.token,
+        )
+
+    def __setstate__(self, state):
```

Comment:
There are ways not to write boilerplate code, like this but it can be optimized later.

Reply:
like using @dataclass, we used in the backend and seems nice (but unnecessary additional dependency) its just minor comment, no need to change.

Reply:
@davidbuniat actually, I'll be putting up a PR soon to remove the use of `@dataclass`. It kills python <= 3.7 compatibility.

Reply:
saw your comment on other review, agree!